assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -24,12 +24,19 @@ import { useEffect, useCallback } from 'react';
  * Internal dependencies
```

Comment:
I think the logic in this file is a little much. Is there any chance you could break it up it's own files / hooks.